Content.Server/GameObjects/Components/Atmos/BarotraumaComponent.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Runtime.CompilerServices;
+using CannyFastMath;
+using Content.Server.GameObjects.Components.Mobs;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Server.Interfaces.GameObjects;
+using Content.Shared.Atmos;
+using Content.Shared.GameObjects;
+using Content.Shared.GameObjects.Components.Mobs;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using NFluidsynth;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.IoC;
+using Robust.Shared.Serialization;
+using Logger = Robust.Shared.Log.Logger;
+
+namespace Content.Server.GameObjects.Components.Atmos
+{
+    /// <summary>
+    ///     Barotrauma: injury because of changes in air pressure.
+    /// </summary>
+    [RegisterComponent]
+    public class BarotraumaComponent : Component
+    {
+        public override string Name => "Barotrauma";
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Update(float frameTime)
+        {
+            if (!Owner.TryGetComponent(out DamageableComponent damageable)) return;
+            Owner.TryGetComponent(out ServerStatusEffectsComponent status);
+
+            var coordinates = Owner.Transform.GridPosition;
+            var gridAtmos = EntitySystem.Get<AtmosphereSystem>().GetGridAtmosphere(coordinates.GridID);
+            var tile = gridAtmos?.GetTile(coordinates);
+
+            var pressure = 1f;
+            var highPressureMultiplier = 1f;
+            var lowPressureMultiplier = 1f;
+
+            foreach (var protection in Owner.GetAllComponents<IPressureProtection>())
+            {
+                highPressureMultiplier *= protection.HighPressureMultiplier;
+                lowPressureMultiplier *= protection.LowPressureMultiplier;
+            }
+
+            if (tile?.Air != null)
+                pressure = MathF.Max(tile.Air.Pressure, 1f);
+
+            switch (pressure)
+            {
+                // Low pressure.
+                case var _ when pressure <= Atmospherics.WarningLowPressure:
```

Comment:
don't miss-use `when` statement like this.

You're binding `pressure` to `_` (the `var _` part), then immediately using `pressure` again...
(And you're only binding pressure to `_` because you HAVE to bind the variable in the switch (`switch(pressure)`) to SOMETHING for a `when` statement to be valid)

i.e.: this line should really be
```cs
case var p when p <= Atmospherics.WarningLowPressure:
```

Ditto for the 'High Pressure' case.

Reply:
Oops, I didn't know that's how you used these... Fixed now, thanks